File: src/main/java/com/freya02/botcommands/internal/prefixed/TextParameterData.java
package com.freya02.botcommands.internal.prefixed;

import com.freya02.botcommands.api.prefixed.annotations.TextOption;
import org.jetbrains.annotations.NotNull;

import java.lang.reflect.Parameter;
import java.util.Optional;

public class TextParameterData {
	private final String optionalName, optionalExample;

	public TextParameterData(Parameter parameter) {
		final TextOption option = parameter.getAnnotation(TextOption.class);

		if (option.name().isBlank()) {
			optionalName = null;
		} else {
			optionalName = option.name();
		}

		if (option.example().isBlank()) {
			optionalExample = null;
		} else {
			optionalExample = option.example();
		}
	}

	@NotNull
	public Optional<String> getOptionalName() {
		return Optional.ofNullable(optionalName);
	}

	@NotNull
	public Optional<String> getOptionalExample() {
		return Optional.ofNullable(optionalExample);
	}
}

File: src/main/java/com/freya02/botcommands/internal/prefixed/TextParameterData.kt
package com.freya02.botcommands.internal.prefixed

import com.freya02.botcommands.annotations.api.prefixed.annotations.TextOption
import java.util.*
import kotlin.reflect.KParameter
import kotlin.reflect.full.findAnnotation

class TextParameterData(parameter: KParameter) {
    private var optionalName: String? = null
    private var optionalExample: String? = null

    init {
        val option = parameter.findAnnotation<TextOption>()!!

        optionalName = option.name.ifBlank { null }
        optionalExample = option.example.ifBlank { null }
    }

    fun getOptionalName(): Optional<String> {
        return Optional.ofNullable(optionalName)
    }

    fun getOptionalExample(): Optional<String> {
        return Optional.ofNullable(optionalExample)
    }
}